src/libraries/System.Text.RegularExpressions/tests/RegexSRMTests.cs

```diff
@@ -91,72 +91,205 @@ public void TestStartSet()
+                File.AppendAllText(outputfile, string.Format("{0}ms, Match:{1} (Index:{2} Length:{3})\n", t1, match1.Success, match1.Index, match1.Length));
+                return t1;
+            }
+            catch (TimeoutException)
+            {
+                File.AppendAllText(outputfile, " TIMEOUT\n");
+                return -1;
+            }
+            catch (Exception)
+            {
+                File.AppendAllText(outputfile, " ERROR\n");
+                return -1;
+            }
+        }
+
+        private static MethodInfo _Deserialize = typeof(Regex).GetMethod("Deserialize", BindingFlags.NonPublic | BindingFlags.Static);
+        private static Regex Deserialize(string s) => _Deserialize.Invoke(null, new object[] { s }) as Regex;
+
+        private static MethodInfo _Serialize = typeof(Regex).GetMethod("Serialize", BindingFlags.NonPublic | BindingFlags.Instance);
+        private static string Serialize(Regex r) => _Serialize.Invoke(r, null) as string;
+
+        private const string serializedout = @"c:\tmp\serialized.txt";
+        /// <summary>
+        /// Test serialization/deserialization and measure performance for all regexes in the regexesfile.
+        /// </summary>
+        //[Fact]
+        private void TestRunSerialization()
+        {
+            string[] rawregexes = File.ReadAllLines(regexesfile);
+            File.AppendAllText(outputfile, string.Format("\n========= TimeStamp:{0} =========\n", System.DateTime.Now));
+            int k = rawregexes.Length;
+            Regex[] rs = new Regex[k];
+            string[] ser = new string[k];
+            Regex[] drs = new Regex[k];
+            //construct
+            int totConstrTime = System.Environment.TickCount;
+            for (int i = 0; i < k; i++)
+                rs[i] = new Regex(rawregexes[i], DFA);
+            totConstrTime = System.Environment.TickCount - totConstrTime;
+            //serialize
+            int totSerTime = System.Environment.TickCount;
+            //repeat ten times
+            for (int j = 0; j < 10; j++)
+                for (int i = 0; i < k; i++)
+                    ser[i] = Serialize(rs[i]);
+            totSerTime = System.Environment.TickCount - totSerTime;
+            totSerTime = totSerTime / 10;
+            //save the serializations
+            File.WriteAllLines(serializedout, ser);
+            //deserialize
+            int totDeSerTime = System.Environment.TickCount;
+            //repeat ten times
+            for (int j = 0; j < 10; j++)
+                for (int i = 0; i < k; i++)
+                    drs[i] = Deserialize(ser[i]);
+            totDeSerTime = System.Environment.TickCount - totDeSerTime;
+            totDeSerTime = totDeSerTime / 10;
+            File.AppendAllText(outputfile, string.Format("\nconstr:{0}ms, serialization:{1}ms, deserialization:{2}ms\n", totConstrTime, totSerTime, totDeSerTime));
```

Comment:
This is more of a diagnostic/debugging aid than a test, maybe worth keeping such things in their own file so that later when we clean up we can decide whether to #if them out or something.

Reply:
Good point! I'll move all my debugging/profiling helper methods into a separate file that can just be deleted later.
I'll do it as the next thing actually. I'll create a new PR to take care of those edits. Currently these are mixed up with the regular unit tests that does create some confusion.
(I also need to add a unit test for detecting timeout of dfa.)